### PR TITLE
improve iso support

### PR DESCRIPTION
**Pull request by ram-mohan**
Test: ./ultrahdr_unit_test

### CODE REVIEW

**Comment by gregbenz** [comment]
These updates not only appear to work properly for ISO encoding, they also seem required for any successful effort with negative boost values for XMP as well (ie, this resolves an important encoding bug in the current mainline).

**Comment by gregbenz** [comment]
While the image works fine in both Adobe and Apple software, the image fails to render in Chrome (including latest Canary). It may be ideal to make ISO support a configurable flag in the demo app.

File: tests/gainmapmetadata_test.cpp [hunk 1]
```diff
@@ -44,12 +44,12 @@ const std::string kIso = "urn:iso:std:iso:ts:21496:-1";
 TEST_F(GainMapMetadataTest, encodeMetadataThenDecode) {
   uhdr_gainmap_metadata_ext_t expected("1.0");
   expected.max_content_boost = 100.5f;
-  expected.min_content_boost = 1.5f;
+  expected.min_content_boost = 0.000578369f;
```
[the remark this answers is not in the document]
Comment (DichenZhang1) [reply tests/gainmapmetadata_test.cpp:47]: Other part looks good to me, but still I will recommend we add a new test case to cover negative values.

[the remark this answers is not in the document]
Comment (DichenZhang1) [reply tests/gainmapmetadata_test.cpp:47]: Yes. We'd better have tests for both positive and negative
